types/vue.d.ts

```diff
@@ -55,12 +56,12 @@ export declare class Vue {
   $delete: typeof Vue.delete;
   $watch(
     expOrFn: string,
-    callback: WatchHandler<this, any>,
+    callback: WatchHandler<any>,
     options?: WatchOptions
   ): (() => void);
   $watch<T>(
     expOrFn: (this: this) => T,
-    callback: WatchHandler<this, T>,
+    callback: WatchHandler<T>,
```

Comment:
Hmm. We cannot reuse WatchHandler because explicit this typing will break ThisType. Just write a new function type will be fine.